video_streaming_ml/face/src/detect_faces.py

```diff
@@ -0,0 +1,106 @@
+# import the necessary packages
+from imutils.video import VideoStream
+from imutils.video import WebcamVideoStream
+from imutils.video import FPS
+import argparse
+import imutils
+import time
+import cv2
+import numpy as np
+import logging
+
+CAFFE_MODEL = 'face/model/res10_300x300_ssd_iter_140000.caffemodel'
+PROTOTEXT = 'face/model/deploy.prototxt.txt'
+SHOW_FRAME = True
+DO_PREDICT = False
+SHOW_FPS = True
+CONFIDENT_THRESHOLD = 0.5
+MAX_NUM_FRAME = 500
+WIDTH = 1000
+
+logging.basicConfig(level=logging.INFO,
+                    format='%(asctime)s :: %(levelname)s :: %(message)s')
+
+# construct the argument parse and parse the arguments
+# ap = argparse.ArgumentParser()
+# ap.add_argument("-c", "--confidence", type=float, default=0.5,
+#                 help="minimum probability to filter weak detections")
+# args = vars(ap.parse_args())
+
+# Load model
+# logging.info("Loading Model")
+net = cv2.dnn.readNetFromCaffe(PROTOTEXT, CAFFE_MODEL)
+
+# initialize the video stream and allow the camera sensor to warm up
+# logging.info("Starting Video Stream")
+vs = VideoStream(src=0).start()
+fps = FPS().start()
+time.sleep(2.0)
+
+# For real-time FPS calculation
+prev_frame_time = 0
+new_frame_time = 0
+
+logging.info("Prediction: {}, Show Frame: {}, Width: {}".format(
+    DO_PREDICT, SHOW_FRAME, WIDTH))
+
+# loop over the frames from the video stream
+while True:
+    if SHOW_FPS:
+        new_frame_time = time.time()
+    # Feed the video stream
+    frame = vs.read()
+
+    # Frame Tranformation
+    frame = imutils.resize(frame, width=WIDTH)
+    # frame = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
+    # frame = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
+    # frame = np.dstack([frame, frame, frame])
+
+    # convert frame to blob
+    (h, w) = frame.shape[:2]
+    blob = cv2.dnn.blobFromImage(cv2.resize(frame, (300, 300)), 1.0,
```

Comment:
Might be helpful to add comments or define these as variables so that it's apparent why 104, 177, 123, etc were chosen here